### PR TITLE
Big heading and search form pattern

### LINKED ISSUE #71: Block Patterns - Search - Big heading and search form
**Description**

🛠️ The goal of this issue is to track the implementation of the `Search - Big heading and search form` pattern. 
🧑‍💻 The implementation must follow the [pattern creation guidelines](https://github.com/WordPress/twentytwentyfive/blob/trunk/README.md#pattern-creation-guidelines), and should be assigned to its appropriate category.
🎨 [Figma design](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3177-1781&t=IqDHO3nwfV13Hcvm-1)

Search_ Big heading and search form

### PR DESCRIPTION
**Pull request by jenniferfarhat**
**Description**
This PR adds a new pattern: Search - Big heading and search form. Also, additional style settings were added to the core/search block on the theme.json.

Additional notes:
[Design](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3177-1781&node-type=SYMBOL&t=9PtDqHUkO22FE08W-0)
[Issue](https://github.com/WordPress/twentytwentyfive/issues/71). 

**Screenshots**

Screenshot 2024-08-30 at 12 42 16 PM

**Testing Instructions**

1. Create a page.
2. Add the Search - big  heading and search form pattern
3. Confirm everything is looking like [the designs](https://www.figma.com/design/dzGCSntVch4EQdVERTqyVK/Twenty-Twenty-Five?node-id=3177-1781&node-type=SYMBOL&t=9PtDqHUkO22FE08W-0).

### CODE REVIEW

File: patterns/search-big-heading.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Categories: call-to-action, text
+ * Description: heading with search form
+ *
+ * @package WordPress
+ * @subpackage Twenty_Twenty_Five
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:group {"metadata":{"name":"Big heading and search form"},"align":"wide","style":{"spacing":{"blockGap":"var:preset|spacing|50","padding":{"top":"140px","bottom":"140px"}}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
```
Comment (shail-mehta) [inline patterns/search-big-heading.php]: @jenniferfarhat Thanks for the PR! 

Could you please remove meta data?

Comment (rejaulalomkhan) [inline patterns/search-big-heading.php]:
```suggestion
<!-- wp:group {"metadata":{"name":"Big heading and search form"},"align":"wide","style":{"spacing":{"blockGap":"var:preset|spacing|50","padding":{"top":"var:preset|spacing|80","bottom":"var:preset|spacing|80"}}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
```

File: patterns/search-big-heading.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+<div class="wp-block-group alignwide" style="padding-top:140px;padding-bottom:140px"><!-- wp:heading {"level":1} -->
```
Comment (rejaulalomkhan) [inline patterns/search-big-heading.php]:
```suggestion
<div class="wp-block-group alignwide" style="padding-top:var(--wp--preset--spacing--80);padding-bottom:var(--wp--preset--spacing--80)"><!-- wp:heading {"level":1} -->
```

**Review by rejaulalomkhan — COMMENTED** [review]
We should use **presets** Instead of hardcod. I'd like to recommend for commit.

image

File: patterns/search-big-heading.php (hunk 1 above)
Comment (jenniferfarhat) [reply patterns/search-big-heading.php]: Thank you for the suggestions @rejaulalomkhan

Comment (jenniferfarhat) [reply patterns/search-big-heading.php]: Metadata has been removed.

File: patterns/search-big-heading.php [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+ * Description: heading with search form
+ *
+ * @package WordPress
+ * @subpackage Twenty_Twenty_Five
+ * @since Twenty Twenty-Five 1.0
+ */
+
+?>
+<!-- wp:group {"align":"wide","style":{"spacing":{"blockGap":"var:preset|spacing|50","padding":{"top":"var:preset|spacing|80","bottom":"var:preset|spacing|80"}}},"layout":{"type":"flex","orientation":"vertical","justifyContent":"stretch"}} -->
+<div class="wp-block-group alignwide" style="padding-top:var(--wp--preset--spacing--80);padding-bottom:var(--wp--preset--spacing--80)"><!-- wp:heading {"level":1} -->
```
Comment (carolinan) [inline patterns/search-big-heading.php]: The main heading or title of the template or page needs to be the only H1, so this block needs to use a different heading level.

On the blog template that shows the full content (the default home template), the page title is an H1 and and the post title is an H2.
So I recommend using H3 here.

Comment (jenniferfarhat) [reply patterns/search-big-heading.php]: Hi @carolinan thank you for your review! I changed the heading from H1 to H3, I noticed on the design pattern that the font size is bigger, should I adjust the font size as well?

Comment (carolinan) [reply patterns/search-big-heading.php]: Earlier today a new font size that matches the one in the design was added to theme.json.
You can pull down the latest changes from trunk to your fork and branch, and use the new size.

There is an example of how to use it here:
https://github.com/WordPress/twentytwentyfive/commit/8b711761eb7433e927b87dd361ce4f2d0ea8e038#diff-0e2183ba9203872d2f674be1dbc4134b15964cdd0cfddf972937f631dd0ce621R16
```
	<!-- wp:paragraph {"align":"center","style":{"typography":{"fontSize":"var(--wp--custom--font-size--huge)"}}} -->
	<p class="has-text-align-center" style="font-size:var(--wp--custom--font-size--huge);">
```
This is only for illustration purpose, you would add it to the heading.

**Comment by beafialho** [comment]
Thank you @jenniferfarhat! When testing on mobile, I noticed the search block goes outside the screen. If we use a group block inside the outer group block, instead of stack it seems to look as intended. I suggest we make that change.

https://github.com/user-attachments/assets/20a4e2f9-13ca-450a-9423-251686a7d184

File: patterns/search-big-heading.php (hunk 3 above)
Comment (jenniferfarhat) [reply patterns/search-big-heading.php]: @carolinan The heading font size has been updated on this [commit](https://github.com/WordPress/twentytwentyfive/pull/166/commits/6470527b81c7eea85414bf5ef5593906ea26557d)

**Comment by jenniferfarhat** [comment]
Hello @beafialho this [commit](https://github.com/WordPress/twentytwentyfive/pull/166/commits/db4659a41b196b933d75d448b71bd601b2a5ef3b) updates the layout per your suggestion, thank you.

**Comment by beafialho** [comment]
> Hello @beafialho this [commit](https://github.com/WordPress/twentytwentyfive/pull/166/commits/db4659a41b196b933d75d448b71bd601b2a5ef3b) updates the layout per your suggestion, thank you.

It's looking good to me!

**Pull request merged**